lib/Sema/CSApply.cpp

```diff
@@ -578,7 +664,7 @@ namespace {
       if (!ref)
         ref = solution.resolveConcreteDeclRef(decl, loc);
 
-      assert(ref.getDecl() == decl);
+      // assert(ref.getDecl() == decl);
```

Comment:
Remnants of a fun debugging session? :)

Reply:
Actually, I think we need to remove this assertion. It seems to be checking that we in fact resolved a `ConcreteDeclRef` for the given decl and not a random one but, I don't think that's a necessary thing to check. If we get a different decl (as we do in this case) I think that's OK. And this is something that will easily be caught via unit tests if there's a logical error.